python/modules/IcePy/Types.cpp

```diff
@@ -3853,13 +3851,6 @@ IcePy::ExceptionWriter::_usesClasses() const
 //
 IcePy::ExceptionReader::ExceptionReader(const ExceptionInfoPtr& info) noexcept : _info(info) {}
 
-IcePy::ExceptionReader::~ExceptionReader()
-{
-    AdoptThread adoptThread; // Ensure the current thread is able to call into Python.
-
-    _ex = 0;
-}
```

Comment:
I think it is safe to remove this because decoding is always done in the Python thread.